### DISCUSSION TITLE
Adding default storage breaks `offliceExchange`s `updates`

### DISCUSSION THREAD

**Discussion by momakes2** [urql-graphql/urql#1369] [Q&A] [answered]
When I add the default storage, my custom updates fail with the error:

```
Invalid undefined: The field at `topicId` is `undefined`, but the GraphQL query expects a scalar (number, boolean, etc) for this field.
(Caused At: "JoinUserDialogMutation" mutation, "dialogInfo" Fragment)
```
However, I've logged the result and it contains all of those. 
![image](https://user-images.githubusercontent.com/71785758/107149342-5696b080-696d-11eb-8d7b-bc7bccd412cc.png)

When I disable the storage or use the simple local storage one, it works.

I have a next js app and I don't need SSR so I haven't added next-urql.

**Comment by kitten** [comment, marked as answer]
Typically this error is issued from `write`, so writing a result to the store when any value is exactly `undefined` (not `null`)

None of this data is read from the storage but this only concerns an update of the cache, so I'm not entirely convinced that it's related to the storage per se. The error also indicates that this happens not during an optimistic mutation but a regular mutation result.

It'd be useful if we could see the custom update, the mutation, and the written data that is causing this though. The document itself in combination with the data should show us exactly which field is the culprit here, since the data alone may not be.
